analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -260,11 +260,24 @@ operation.Kind switch
         private static ExceptionState ExceptionCandidate(IOperationWrapperSonar operation) =>
             operation.Instance.Kind switch
             {
+                OperationKindEx.ArrayElementReference => ExceptionState.UnknownException, // ToDo: be explicit regarding exception type
+                OperationKindEx.Conversion => operation.IsImplicit ? null : ExceptionState.UnknownException, // ToDo: be explicit regarding exception type
```

Comment:
Let's also recognize narrowing vs. widening and do not throw for widening (if I haven't mixed those up). Basically
```
var x = (BaseType)concreteType; // Should not throw
var y = (ConcreteType)obj;     // Should be candidate
```